File: 2255-count-prefixes-of-a-given-string/2255-count-prefixes-of-a-given-string.py
class Solution:
    def countPrefixes(self, words: List[str], s: str) -> int:
        # c = 0
        # for i in words:
        #     if s.find(i) == 0:
        #         c += 1
        # return c
        return len(list(filter(lambda w: s.startswith(w), words)))

File: 2255-count-prefixes-of-a-given-string/2255-count-prefixes-of-a-given-string.java
class Solution {
    public int countPrefixes(String[] words, String s) {
        int c = 0, slen = s.length();
        for(int i = 0; i < words.length; i++){
            if(words[i].length() > slen)
                continue;
            boolean flag = true;
            for(int j = 0; j < words[i].length(); j++){
                if(words[i].charAt(j) != s.charAt(j)){
                        flag = false;
                        break;
                }
            }
            if(flag)
                c++; 
        }
        return c;
    }
}